dotCMS/src/integration-test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPITest.java

```diff
@@ -264,4 +273,68 @@ public void test_deleteLanguage_WithExistingContent_ShouldFail() {
 		}
 	}
 
+
+	/**
+	 * Basically we test two methods in conjunction {@link LanguageAPI#makeDefault(Long, User)} and {@link LanguageAPI#transferAssets(Long, Long, User)}
+	 * Given scenario: Create 2 Languages besides the default. Then Create content under the default and switch to a new language. And Call the transferAssets method 
+	 * Expected Result: The api should be able to switch to a new lang and transfer all assets under the old lang to a new default lang.
+	 *
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 * @throws DotIndexException
+	 */
+	@Test
+	public void Create_Content_Under_Default_Lang_Make_New_Default_Language_And_Test_Assets_Lang_Transfer()
+			throws DotDataException, DotSecurityException, DotIndexException {
+		final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+		final Language defaultLang = languageAPI.getDefaultLanguage();
+		final Language newDefaultLanguage = new LanguageDataGen().nextPersisted();
+		final Language thirdLanguage = new LanguageDataGen().nextPersisted();
+
+		final ContentType news = getNewsLikeContentType("News");
+
+		final Contentlet persistedWithOldDefaultLang = new ContentletDataGen(news)
+				.languageId(defaultLang.getId())
+				.setProperty("title", "News Test")
+				.setProperty("urlTitle", "news-test").setProperty("byline", "news-test")
+				.setProperty("sysPublishDate", new Date()).setProperty("story", "news-test")
+				.nextPersisted();
+
+		final Contentlet persistedWithThirdLang = new ContentletDataGen(news)
+				.languageId(thirdLanguage.getId())
+				.setProperty("title", "News Test")
+				.setProperty("urlTitle", "news-test").setProperty("byline", "news-test")
+				.setProperty("sysPublishDate", new Date()).setProperty("story", "news-test")
+				.nextPersisted();
+
+
+	  final User admin = mockAdminUser();
+      languageAPI.makeDefault(newDefaultLanguage.getId(), admin);
```

Comment:
are you switching back the default language to English? I'll do it in a finally fragment

Reply:
I am now.